frontend/packages/ceph-storage-plugin/src/utils/install.ts

```diff
@@ -48,10 +52,13 @@ export const getTotalDeviceCapacity = (list: Discoveries[]): number =>
 
 export const getAssociatedNodes = (pvs: K8sResourceKind[]): string[] => {
   const nodes = pvs.reduce((res, pv) => {
-    const nodeName = pv?.metadata?.labels?.[HOSTNAME_LABEL_KEY];
-    if (nodeName) {
-      res.add(nodeName);
-    }
+    const matchExpressions: MatchExpression[] =
+      pv?.spec?.nodeAffinity?.required?.nodeSelectorTerms?.[0]?.matchExpressions;
```

Comment:
what if there are multiple node selectors? I think that was the reason we relied on PV label.

Reply:
For LSO PVs, it will always have hostname based selector.
I confirmed with LSO team on that.

Reply:
Relying on node affinity is more reliable over label selector. The node affinity field is a mandatory one for PV and will always be present on every PV unlike labels - added as per convenience.
Thats why we had issues in supporting 4.5 OCS on OCP 4.6 where users were trying to create storage cluster via local volume CR and they are blocked, since the pVs created via local volume are not labeled by hostname